src/main/java/org/openlmis/fulfillment/extension/point/UuidOrderNumberGenerator.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *  
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. 
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org. 
+ */
+
+package org.openlmis.fulfillment.extension.point;
+
+import org.openlmis.fulfillment.domain.Order;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+/**
+ * A default implementation of OrderNumberGenerator extension that generates random UUIDs.
+ * @see OrderNumberGenerator
+ */
+@Component("UuidOrderNumberGenerator")
+public class UuidOrderNumberGenerator implements OrderNumberGenerator {
+
+  /**
+   * Generates unique number for given order.
+   */
+  public String generate(Order order) {
+    return UUID.randomUUID().toString();
```

Comment:
If we look at the existing code, we didn't just generate another random UUID.  Instead we used the [UUID from Order.getExternalId()](https://github.com/OpenLMIS/openlmis-fulfillment/blob/master/src/main/java/org/openlmis/fulfillment/domain/OrderNumberConfiguration.java#L82).

Talking with @koch0142 however I think we can make this even simpler, OpenLMIS would also like a shorter Order code / number.  Here's what we'd propose:
- Rename this class to be called Base36EncodedOrderNumberGenerator
- Instead of returning a UUID, take Order.getId(), and encode it as base36.
- Truncate the base36 encoding to 8 characters.

I think this will work for everyone.  We'll keep the good work you've done here to introduce the extension point, and if some other group wants to leverage it, they may.  The default implementation however will be to start using a 8 character base36 encoding of the Order ID (not the external id).